plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -1765,7 +1765,7 @@ protected ScaleVmAnswer execute(ScaleVmCommand cmd) {
 
             // Check if license supports the feature
             VmwareHelper.isFeatureLicensed(hyperHost, FeatureKeyConstants.HOTPLUG);
-            VmwareHelper.setVmScaleUpConfig(vmConfigSpec, vmSpec.getCpus(), vmSpec.getMaxSpeed(), vmSpec.getMinSpeed(), (int) requestedMaxMemoryInMb, ramMb,
+            VmwareHelper.setVmScaleUpConfig(vmConfigSpec, vmSpec.getCpus(), vmSpec.getMaxSpeed(), getReservedCpuMHZ(vmSpec), (int) requestedMaxMemoryInMb, ramMb,
```

Comment:
How about min of both min speed and reserved mhz?

Reply:
@rhtyd `getReservedCpuMHZ()` method seems to be using min CPU speed itself based CPU reservation global setting. 

https://github.com/apache/cloudstack/blob/4.16/plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java#L3052-L3057

Same method is used while deploying VM, https://github.com/apache/cloudstack/blob/4.16/plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java#L1999

Reply:
yes getReservedCpuMHZ() makes sense, since it will set 0 if VmwareReserveCPU is false which is correct.